app/controllers/concerns/contacts_controllers.rb

```diff
@@ -0,0 +1,19 @@
+class ContactsController < ApplicationController
+
+  def create
+    @company = find_company(params[:company_id])
+    contact = @company.contacts.new(contacts_params)
+    if contact.save
+      flash[:success] = "#{contact.name} was added as a contact to #{@company.name}"
+      redirect_to company_path(@company)
+    else
+      flash[:danger] = "Contact must have a Name and Email to be valid."
+      redirect_to company_path(@company)
```

Comment:
if a user doesn't enter the required info, then we should probably render the contact new pages again (or in this case whatever view has that form)